client/state/sharing/reducer.js

```diff
@@ -2,13 +2,24 @@
  * Internal dependencies
  */
 
+import { CONNECTIONS_EXPANDED_SERVICE } from 'state/action-types';
 import keyring from './keyring/reducer';
-import { combineReducers } from 'state/utils';
+import { combineReducers, withoutPersistence } from 'state/utils';
 import publicize from './publicize/reducer';
 import services from './services/reducer';
 
+export const expandedService = withoutPersistence( ( state = '', action ) => {
+	switch ( action.type ) {
+		case CONNECTIONS_EXPANDED_SERVICE:
+			return action.serviceName ? action.serviceName : null;
```

Comment:
This threw an error if action.serviceName was undefined, eg. if just going direct to http://calypso.localhost:3000/marketing/connections/ without a service name in the path - apparently reducers need to return null instead of undefined - so making this change fixed it.